commit d91b4a300017bdcfbea8d013f05369ebba9a3d1e
Author: Cameron Nemo <cam@nohom.org>
Date:   Sun Aug 7 11:10:31 2022 -0700

    meson: add option to force docbook2x format
    
    Some distros ship docbook2x as docbook2man, which perplexes the logic in
    the meson.build that chooses the "docdtd" value.
    
    Add a build option to always use the newer docbook2x docdtd value.
    
    Signed-off-by: Cameron Nemo <cam@nohom.org>

diff --git a/meson.build b/meson.build
index a145faf06..9c0e6e488 100644
--- a/meson.build
+++ b/meson.build
@@ -135,6 +135,7 @@ cgrouppattern = get_option('cgroup-pattern')
 coverity = get_option('coverity-build')
 init_script = get_option('init-script')
 sanitize = get_option('b_sanitize')
+docbook2x_only = get_option('docbook2x-only')
 want_examples = get_option('examples')
 want_io_uring = get_option('io-uring-event-loop')
 want_pam_cgroup = get_option('pam-cgroup')
@@ -337,7 +338,7 @@ docconf.set('LXCTEMPLATEDIR', lxctemplatedir)
 docconf.set('LXC_USERNIC_CONF', lxc_user_network_conf)
 docconf.set('LXC_USERNIC_DB', lxc_user_network_db)
 docconf.set('PACKAGE_VERSION', version_data.get('LXC_VERSION'))
-if sgml2man.found() and docbook2man.found() and sgml2man.full_path() == docbook2man.full_path()
+if sgml2man.found() and docbook2man.found() and sgml2man.full_path() == docbook2man.full_path() and not docbook2x_only
     docconf.set('docdtd', '"-//Davenport//DTD DocBook V3.0//EN"')
 else
     docconf.set('docdtd', '"-//OASIS//DTD DocBook XML" "http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd"')
diff --git a/meson_options.txt b/meson_options.txt
index c14dacf27..ccc4dde5d 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -9,6 +9,10 @@ option('cgroup-pattern', type: 'string', value: '',
 option('coverity-build', type: 'boolean', value: 'true',
        description: 'build for coverity')
 
+# no configure equivalent
+option('docbook2x-only', type: 'boolean', value: 'false',
+       description: 'always use DocBook 2x format')
+
 # was --{disable,enable}-examples in autotools
 option('examples', type: 'boolean', value: 'true',
        description: 'build and install examples')
